app/src/main/java/sw19/moring03/paint/MainActivity.java

```diff
@@ -139,4 +156,31 @@ public void chooseNewColor(View view) {
 
         colorChooserMenu.dismiss();
     }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == CAMERA_REQUEST) {
+            try {
+                Bitmap cameraPicture = (Bitmap) data.getExtras().get("data");
+
+                if (cameraPicture == null) {
+                    Toast toast = Toast.makeText(getApplicationContext(),
+                            "Could not load image from Camera!", Toast.LENGTH_SHORT);
+                    toast.show();
+                    return;
+                }
+                newPhoto = cameraPicture;
+                setChosenTool(Tool.TAKE_PHOTO);
+
+            } catch (Exception e) {
+                Toast toast = Toast.makeText(getApplicationContext(),
```

Comment:
Leave the stacktrace and remove the toast. I'll write a littlle logger class by time